### PR TITLE
260861: Retrieve Access Token for DB Migrations using Workload Identity

### PR DESCRIPTION
**Pull request by jeevankuduvaravindran**
This PR contains the changes to Retrieve Access Token for DB Migrations using Workload Identity

[AB#260861](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/260861)

FinallyAnetraGIF

### CODE REVIEW

File: db-migration.Dockerfile [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
+
+# Install Azure CLI
+RUN curl -skL https://aka.ms/InstallAzureCLIDeb | bash
+
+# Copy scripts and migration changelog files
+COPY --chown=liquibase:liquibase --chmod=755 changelog ./changelog
+COPY --chown=liquibase:liquibase --chmod=755 scripts ./scripts
+
+# Get AAD access token and run database migration
+CMD ["sh", "-c", "az login --service-principal --username $AZURE_CLIENT_ID --federated-token $(cat $AZURE_FEDERATED_TOKEN_FILE) --tenant $AZURE_TENANT_ID; ACCESS_TOKEN=$(az account get-access-token --resource-type oss-rdbms --query 'accessToken' --output tsv); export ACCESS_TOKEN; cd scripts/migration; sh database-up $ACCESS_TOKEN; exit 0;"]
```
Comment (ppotturi) [inline db-migration.Dockerfile:15]: Do we have to do this for all services? Can we have it as a common logic?

Comment (jeevankuduvaravindran) [reply db-migration.Dockerfile]: @ppotturi  We could have this taken out as a separate script file. But we might still need to add that script file to all applications that need migration.

Comment (ppotturi) [reply db-migration.Dockerfile]: As we discussed, We can keep this logic here. But we will investigate if there are any better ways to do it.

**Review by ppotturi — APPROVED** [review]

File: db-migration.Dockerfile [hunk 2]
```diff
@@ -1,4 +1,15 @@
-FROM liquibase/liquibase:3.10.x
+# Use Liquibase as base image
+FROM liquibase/liquibase:4.25
 
-COPY --chown=liquibase:liquibase changelog ./changelog
-COPY --chown=liquibase:liquibase scripts /scripts
\ No newline at end of file
+# Change to root user for installation
+USER root
```
Comment (kbabigumira) [inline db-migration.Dockerfile:5]: Won't AKS policy block this running with elevated user access?
@ppotturi  @jeevankuduvaravindran  @ajaysoni-cap

**Review by kbabigumira — APPROVED** [review]

File: db-migration.Dockerfile (hunk 2 above)
Comment (jeevankuduvaravindran) [reply db-migration.Dockerfile:5]: @kbabigumira , As discussed, we can have a look at this during policy implementation

**Pull request merged**